spec/models/post_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  before(:all) do
+    @user = User.create(
+      name: 'Loman',
+      bio: 'Software Developer',
+      photo: 'http://hey.com/og.png',
+      posts_counter: 0
+    )
+    @post = Post.create(
+      title: 'Hello',
+      text: 'Software Developer',
+      author_id: @user.id,
+      comments_counter: 0,
+      likes_counter: 0
+    )
+  end
+
+  it 'post title should be present' do
+    @post.title = nil
+    expect(@post).to_not be_valid
+  end
+
+  it 'A posts likes_counter should be >= 0' do
+    @post.likes_counter = -4
+    expect(@post).to_not be_valid
+  end
+
+  it 'A posts comments_counter should be >= 0' do
+    @post.comments_counter = -4
+    expect(@post).to_not be_valid
+  end
+
+  it 'A posts likes_counter should be numeric' do
+    @post.likes_counter = 'One'
+    expect(@post).to_not be_valid
+  end
+
+  it 'should not accept more than 250 character' do
+    @post.title = '
+    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula
+    eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient
+    montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque
+    eu, pretium quis,'
+    expect(@post).to_not be_valid
+  end
+  it 'fetch_recent_comments' do
+    @user.save
+    @post.save
+    @comment = Comment.create(text: 'Hello World!', author_id: @user.id, post_id: @post.id)
+    @comment = Comment.create(text: 'Welcome World!!', author_id: @user.id, post_id: @post.id)
+    expect(@post.fetch_recent_comments.count).to eq(2)
+  end
```

Comment:
- [ ] Great job so far 👏🏻 However, all custom methods in the models must have a corresponding unit test. Actually, you are missing a test to test the `update_posts_counter` method of your Post model, please add this ✍🏻

```ruby
  def update_posts_counter
     author.increment!(:posts_counter)
  end
```